server-tasks/src/main/java/org/infinispan/tutorial/simple/server/tasks/InfinispanServerTasks.java

```diff
@@ -1,122 +1,59 @@
-      parameters.put("name", "developer");
+      parameters.put("greetee", "developer");
 
       // Execute hello task
       String greet = execCache.execute("hello-task", parameters);
       System.out.printf("Greeting = %s\n", greet);
 
-      // Store some values and compute the sum
-      int range = 10;
-      IntStream.range(0, range).boxed().forEach(
-         i -> dataCache.put(i + "-key", i + "-value")
-      );
-      int result = execCache.execute("sum-values-task", Collections.emptyMap());
-      System.out.printf("Sum of values = %d\n", result);
-
       // Stop the cache manager and release all resources
-      dataCache.getRemoteCacheManager().stop();
       execCache.getRemoteCacheManager().stop();
    }
 
-   public static RemoteCache<String,String> getDataCache() {
-      // Create a configuration for a locally-running server
-      ConfigurationBuilder builder = new ConfigurationBuilder();
-      builder
-         .addServer().host("127.0.0.1").port(11222)
-         .marshaller(UTF8StringMarshaller.class);
+   private static void uploadTask() throws IOException, InterruptedException {
+      String taskPostUrl = String.format("http://localhost:%d/rest/v2/tasks/hello-task", 11222);
+      String script = getResourceAsString("hello.js", InfinispanServerTasks.class.getClassLoader());
+      HttpClient client = new HttpClient();
+      PostMethod postMethod = new PostMethod(taskPostUrl);
+      postMethod.setRequestHeader("Content-type", APPLICATION_JAVASCRIPT_TYPE);
 
-      RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build());
-      final RemoteCache<String, String> cache = cacheManager.getCache();
-      cache.clear();
-      return cache;
+      postMethod.setRequestEntity(new StringRequestEntity(script, null, null));
+      client.executeMethod(postMethod);
    }
 
-   public static RemoteCache<String,String> getExecCache() {
+   private static RemoteCache<String,String> getExecCache() {
       ConfigurationBuilder builder = new ConfigurationBuilder();
       builder.addServer().host("127.0.0.1").port(11222);
 
       RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build());
-      return cacheManager.getCache();
-   }
-
-   public static class HelloTask implements ServerTask<String> {
-
-      private TaskContext ctx;
-
-      @Override
-      public void setTaskContext(TaskContext ctx) {
-         this.ctx = ctx;
-      }
-
-      @Override
-      public String call() throws Exception {
-         String name = (String) ctx.getParameters().get().get("name");
-         return "Hello " + name;
-      }
-
-      @Override
-      public String getName() {
-         return "hello-task";
-      }
-
+      return cacheManager.administration()
+              .withFlags(CacheContainerAdmin.AdminFlag.VOLATILE)
+              .getOrCreateCache("data", "org.infinispan.DIST_SYNC");
    }
-
-   public static class SumValuesTask implements ServerTask<Integer> {
```

Comment:
can we keep this task instead? it seems a better example than "hello world"

Reply:
Because it needs to be deployed in the server with a jar

Reply:
these are simple tutorials